bakery/discharge.go

```diff
@@ -5,18 +5,36 @@ import (
 
 	"gopkg.in/errgo.v1"
 	"gopkg.in/macaroon.v1"
+
+	"gopkg.in/macaroon-bakery.v1/bakery/checkers"
 )
 
-// DischargeAll gathers discharge macaroons for all the third party caveats
-// in m (and any subsequent caveats required by those) using getDischarge to
-// acquire each discharge macaroon.
-// It returns a slice with m as the first element, followed by
-// all the discharge macaroons. All the discharge macaroons
-// will be bound to the primary macaroon.
+// DischargeAll gathers discharge macaroons for all the third party
+// caveats in m (and any subsequent caveats required by those) using
+// getDischarge to acquire each discharge macaroon. It returns a slice
+// with m as the first element, followed by all the discharge macaroons.
+// All the discharge macaroons will be bound to the primary macaroon.
 func DischargeAll(
 	m *macaroon.Macaroon,
 	getDischarge func(firstPartyLocation string, cav macaroon.Caveat) (*macaroon.Macaroon, error),
 ) (macaroon.Slice, error) {
+	return DischargeAllWithKey(m, getDischarge, nil)
+}
+
+// DischargeAllWithKey is like DischargeAll except that the selfKey
+// parameter may optionally hold the key of the client, in which case it
+// will be used to discharge any third party caveats with the special
+// location "self". In this case, the caveat itself must be "true". This
```

Comment:
"client" was going to be my suggestion. I can't think of anything really good, but I think client makes it clear when minting and discharging.